### PR TITLE
partial concurrent for DeliverTxs

### PR DESCRIPTION
**Pull request by MinZhu09**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/tendermint/state/execution.go [hunk 1]
```diff
@@ -287,9 +312,14 @@ func (blockExec *BlockExecutor) runAbci(block *types.Block, deltaInfo *DeltaInfo
 				db:       blockExec.db,
 				proxyApp: blockExec.proxyApp,
 			}
-			if blockExec.isAsync {
+			switch blockExec.deliverTxsExecMode {
```
Comment (scf0220) [inline libs/tendermint/state/execution.go]: preRun also should modify

File: app/repair_state.go [hunk 2]
```diff
@@ -162,7 +162,8 @@ func doRepair(ctx *server.Context, state sm.State, stateStoreDB dbm.DB,
 	err = startEventBusAndIndexerService(ctx.Config, eventBus, ctx.Logger)
 	panicError(err)
 	blockExec := sm.NewBlockExecutor(stateStoreDB, ctx.Logger, proxyApp.Consensus(), mock.Mempool{}, sm.MockEvidencePool{})
-	blockExec.SetIsAsyncDeliverTx(viper.GetBool(sm.FlagParalleledTx))
+	//blockExec.SetIsAsyncDeliverTx(viper.GetBool(sm.FlagParalleledTx))
```
Comment (scf0220) [inline app/repair_state.go]: delete it

File: libs/cosmos-sdk/baseapp/baseapp_mode_base.go [hunk 3]
```diff
@@ -36,6 +36,8 @@ func (app *BaseApp) getModeHandler(mode runTxMode) modeHandler {
 		h = &modeHandlerSimulate{&modeHandlerBase{mode: mode, app: app}}
 	case runTxModeDeliverInAsync:
 		h = &modeHandlerDeliverInAsync{&modeHandlerBase{mode: mode, app: app}}
+	case runTxModeDeliverPartConcurrent:
```
Comment (scf0220) [inline libs/cosmos-sdk/baseapp/baseapp_mode_base.go]: use runTxModeDeliver?

File: libs/cosmos-sdk/baseapp/baseapp_partial_concurrent.go [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
+func (dttr *dttRoutine) makeNewTask(txByte []byte, index int) {
+	dttr.step = dttRoutineStepStart
+	dttr.txIndex = index
+	dttr.needToRerun = false
+	dttr.txByte <- txByte
+}
+
+func (dttr *dttRoutine) start() {
+	dttr.done = make(chan int8)
+	dttr.txByte = make(chan []byte, 3)
```
Comment (scf0220) [inline libs/cosmos-sdk/baseapp/baseapp_partial_concurrent.go]: use variables？

File: libs/cosmos-sdk/baseapp/baseapp_partial_concurrent.go [hunk 5]
```diff
@@ -0,0 +1,665 @@
...
+	dttr.rerunCh = make(chan int8, 5)
```
Comment (scf0220) [inline libs/cosmos-sdk/baseapp/baseapp_partial_concurrent.go]: use variables？

File: libs/cosmos-sdk/baseapp/baseapp_partial_concurrent.go [hunk 6]
```diff
@@ -0,0 +1,659 @@
...
+		return
+	}
+
+	dttm.done = make(chan int8, 1)
+	dttm.txs = txs
+	dttm.serialTask = nil
+	dttm.serialIndex = -1
+	dttm.serialCh = make(chan int8, 4)
+
+	dttm.checkStateCtx = dttm.app.checkState.ctx
```
Comment (MinZhu09) [inline libs/cosmos-sdk/baseapp/baseapp_partial_concurrent.go:265]: use deliverState

**Review by scf0220 — APPROVED** [review]

**Pull request merged**